### PR TITLE
Add import package

### PR DESCRIPTION
**Pull request by zaerl**
This PR adds the Import package and loads it inside the Jetpack plugin.

In this conversation p1675062364389839-slack-CDLH4C1UZ, we decided it is better to move [previous plugin codebase](https://github.com/Automattic/jetpack/tree/add/import-plugin) to a composer package.

Project details: pcmemI-1ro-p2
Additional details: pcmemI-1Dn-p2

## Proposed changes:

* Moved the REST endpoints from the Import plugins to the Import package and refactored them.
* Include the Import package in the list of packages in the `Config` package.
* Include the Import package in the `Jetpack` package so that the REST endpoints are automatically loaded for all users.
* Remove additional loading of `Config` packages.
* Change the REST endpoints `namespace` from `jetpack/v4` to `jetpack/v4/import`. In WPCOM we have a new check that generates an error if the package is not loaded. See D100594-code.
* Add check for duplicated posts.
* Add support for posts with the type `page`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Detailed instructions: pcmemI-1LC-p2.

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
I'm thinking you may be able to simplify things a bit here. Since the package will be used in the Jetpack plugin, where you've used the Config package to initialize the Import package, I don't believe you need to worry about the Config package inside your own package.

I don't think you need to worry about Sync either, since the Jetpack plugin takes care of that for you.

As a result, something like this may be enough (I'm happy to push it to the branch if that's easier for you):

```diff
diff --git a/projects/packages/import/composer.json b/projects/packages/import/composer.json
index 6213951c51..885ba0b48d 100644
--- a/projects/packages/import/composer.json
+++ b/projects/packages/import/composer.json
@@ -4,8 +4,6 @@
 	"type": "jetpack-library",
 	"license": "GPL-2.0-or-later",
 	"require": {
-		"automattic/jetpack-config": "@dev",
-		"automattic/jetpack-sync": "@dev",
 		"automattic/jetpack-connection": "@dev"
 	},
 	"require-dev": {
diff --git a/projects/packages/import/src/class-main.php b/projects/packages/import/src/class-main.php
index df7a564ae3..44bb64e232 100644
--- a/projects/packages/import/src/class-main.php
+++ b/projects/packages/import/src/class-main.php
@@ -7,36 +7,12 @@
 
 namespace Automattic\Jetpack\Import;
 
-use Automattic\Jetpack\Config;
 use Automattic\Jetpack\Connection\Rest_Authentication;
-use Automattic\Jetpack\Sync\Data_Settings;
 
 /**
  * This class will provide endpoint for the Unified Importer.
  */
 class Main {
-
-	/**
-	 * Slug.
-	 *
-	 * @var string
-	 */
-	const PACKAGE_SLUG = 'jetpack-import';
-
-	/**
-	 * Package name.
-	 *
-	 * @var string
-	 */
-	const PACKAGE_NAME = 'Jetpack Import';
-
-	/**
-	 * Package URL.
-	 *
-	 * @var string
-	 */
-	const PACKAGE_URI = 'https://jetpack.com/jetpack-import';
-
 	/**
 	 * Package version.
 	 *
@@ -56,38 +32,26 @@ class Main {
 	 *
 	 * @var \WP_REST_Controller[]
 	 */
-	private $routes = array();
-
-	/**
-	 * Class constructor.
-	 *
-	 * @return void
-	 */
-	private function __construct() {
-	}
+	private static $routes = array();
 
 	/**
-	 * Before everything else starts getting initalized, we need to initialize Jetpack using the
-	 * Config object.
+	 * Constructor.
 	 *
 	 * @return void
 	 */
 	public static function configure() {
-		$import = new self();
-		$config = new Config();
-
-		$config->ensure(
-			'connection',
-			array(
-				'slug'     => self::PACKAGE_SLUG,
-				'name'     => self::PACKAGE_NAME,
-				'url_info' => self::PACKAGE_URI,
-			)
-		);
+		if ( did_action( 'jetpack_import_initialized' ) ) {
+			return;
+		}
 
-		$config->ensure( 'sync', Data_Settings::MUST_SYNC_DATA_SETTINGS );
+		add_action( 'rest_api_init', array( __CLASS__, 'initialize_rest_api' ) );
 
-		add_action( 'rest_api_init', array( $import, 'initialize_rest_api' ) );
+		/**
+		 * Runs right after the Jetpack Import package is initialized.
+		 *
+		 * @since $$next-version$$
+		 */
+		do_action( 'jetpack_import_initialized' );
 	}
 
 	/**
@@ -95,7 +59,7 @@ class Main {
 	 *
 	 * @return void
 	 */
-	public function initialize_rest_api() {
+	public static function initialize_rest_api() {
 		// Set up the REST authentication hooks.
 		Rest_Authentication::init();
 
@@ -106,11 +70,17 @@ class Main {
 			'tags'       => new Endpoints\Tag(),
 		);
 
-		// Allow other plugins to modify import routes.
-		$this->routes = apply_filters( 'jetpack_import_types', $routes );
+		/**
+		 * Allow other plugins to modify import routes.
+		 *
+		 * @since $$next-version$$
+		 *
+		 * @param array $routes Array of import routes.
+		 */
+		self::$routes = apply_filters( 'jetpack_import_types', $routes );
 
 		// Register all the routes.
-		foreach ( $this->routes as $route ) {
+		foreach ( self::$routes as $route ) {
 			$route->register_routes();
 		}
 	}
```

File: projects/packages/import/src/class-main.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	 * Config object.
+	 *
+	 * @return void
+	 */
+	public static function configure() {
+		if ( did_action( 'jetpack_import_initialized' ) ) {
+			return;
+		}
+
+		add_action( 'rest_api_init', array( __CLASS__, 'initialize_rest_api' ) );
```
Comment (jeherve) [inline projects/packages/import/src/class-main.php]: Should we check that a site is connected to WordPress.com before we add the routes and do anything? 

Maybe we could rely on the Connection package we have at our disposal, and bail early in the package if some scenarios?

Comment (zaerl) [reply projects/packages/import/src/class-main.php]: Of course. These endpoints are used by the new Unified Importer and must be accessed only by a user that can `import` and that is connected to WordPress.com.

**Comment by zaerl** [comment]
I don't know why the E2E failed. I will re-run it again.

File: projects/packages/import/src/endpoints/class-post.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+				__( 'Cannot create existing post.', 'jetpack-import' ),
+				array(
+					'status'  => 400,
+					'post_id' => $post_id,
+				)
+			);
+		}
+
+		// WXR saves terms as slugs, so we need to convert them to IDs before send the data to the legacy endpoint.
+		foreach ( array( 'categories', 'tags' ) as $taxonomy ) {
```
Comment (jeroenpf) [inline projects/packages/import/src/endpoints/class-post.php:94]: Perhaps this can be extracted into its own method.

**Review by jeroenpf — CHANGES_REQUESTED** [review]
Just a small comment to extract some logic into its own method... Other than that, it looks good from my point of view.

File: projects/packages/import/src/endpoints/class-post.php (hunk 2 above)
Comment (zaerl) [reply projects/packages/import/src/endpoints/class-post.php:94]: Of course, it's better. Moved.

**Review by jeroenpf — DISMISSED** [review]
LGTM :-)

**Review by jeroenpf — DISMISSED** [review]

**Review by jeherve — DISMISSED** [review]

**Review by bindlegirl — APPROVED** [review]

**Comment by samiff** [comment]
@zaerl @jeroenpf 

Just a heads up that this change will be deployed to WoA sites likely tomorrow in Jetpack version 11.9-a.5 (I'll try to ping you once the changes are live on wpcom/Atomic).

I just wanted to make sure that you had a chance to test these changes on a WoA site specifically. From the testing instructions I saw a JN site mentioned which isn't quite the same as a WoA site, but maybe I misunderstood.

**Comment by zaerl** [comment]
@samiff @jeroenpf Of course. You can see detailed instructions for WoA on our CfT pdtkmj-1cT-p2

**Comment by samiff** [comment]
> @samiff @jeroenpf Of course. You can see detailed instructions for WoA on our CfT pdtkmj-1cT-p2

Thanks, that link was easier to follow along with 👍 Jetpack 11.9-a.5 with this change is now live on simple/WoA sites.

**Pull request merged**